### PR TITLE
Use new class ProxyConfig for proxy selection in OIDC configuration

### PR DESCRIPTION
**Pull request by zgael**
Hi,

following this (closed/not merged) issue https://github.com/stevespringett/Alpine/pull/422 , I create this new one to make use of the newly created ProxyConfig class in order to have OIDC configuration use proxy if declared in configuration.

Thanks @nscuro for the work on ProxyConfig, looks good to me !

### CODE REVIEW

**Review by nscuro — COMMENTED** [review]
Thanks @zgael! 

Code looks good to me, let me test it with DT to make sure it works as intended.

**Comment by zgael** [comment]
Depending on how you plan to test it, I might be able to help : 
- if that means you will build a docker image for DT (with a specific tag) and push it somewhere public, I could pull it and try to use it as I originally intended to (behind corporate proxy) with minimal effort (as I would reinvest that time when deploying it).
- if you were by some way planning to build and test things locally, as I'm in no way familiar with building DT, I'm not sure I will be fit for this task. But once a newer version of DT is released, I would perform the test anyway, so you'd have feedback.

Keep me informed!

**Comment by stevespringett** [comment]
@nscuro Have you tested this with DT?

**Comment by nscuro** [comment]
@stevespringett Not yet, sorry. I'll try to get it done tomorrow.

File: alpine-server/src/main/java/alpine/server/auth/OidcUserInfoAuthenticator.java [hunk 1]
```diff
@@ -42,10 +47,13 @@ class OidcUserInfoAuthenticator {
...
         final UserInfoResponse userInfoResponse;
         try {
-            final var httpResponse =
-                    new UserInfoRequest(configuration.getUserInfoEndpointUri(), new BearerAccessToken(accessToken))
-                            .toHTTPRequest()
-                            .send();
+            HTTPRequest httpRequest = new UserInfoRequest(configuration.getUserInfoEndpointUri(), new BearerAccessToken(accessToken)).toHTTPRequest();
+            final ProxyConfig proxyCfg = ProxyUtil.getProxyConfig();
+
+            if (proxyCfg.shouldProxy(configuration.getUserInfoEndpointUri().toURL())) {
```
[the remark this answers is not in the document]
Comment (nscuro) [reply alpine-server/src/main/java/alpine/server/auth/OidcUserInfoAuthenticator.java]: @zgael A check for `proxyCfg` being `null` would be good here. `getProxyConfig` will return `null` when no proxy config was found at all.

[the remark this answers is not in the document]
Comment (nscuro) [reply alpine-server/src/main/java/alpine/server/auth/OidcUserInfoAuthenticator.java]: Oh, was already addressed. Never mind then, resolving this finding.

**Review by nscuro — APPROVED** [review]
Tested with Dependency-Track by proxying requests to a local Keycloak server through ZAP. Both config resolution and UserInfo requests are proxied correctly for the `ALPINE_HTTP_PROXY_*` and "standard" `HTTP_PROXY` way. Including the Keycloak address in `NO_PROXY` bypasses the proxy as expected.

Screenshot 2023-02-26 at 12 57 55

This is good to merge from my side @stevespringett. Thanks for the PR @zgael! 🚀

**Pull request merged**